utils/update-next.js

```diff
@@ -1,55 +1,133 @@
 
 function exec(cmd, args, opts) {
-  // eslint-disable-next-line no-console
-  console.log({ input: `${cmd} ${args.join(' ')}` });
+  console.log({ input: `${cmd} ${args.join(' ')}`, cwd: opts?.cwd });
   const output = execFileSync(cmd, args, opts).toString().trim();
   console.log({ output });
   console.log();
   return output;
 }
 
-module.exports = async ({ github, context }) => {
-  const oldVersion = require('../examples/nextjs/package.json').dependencies
-    .next;
+module.exports = async ({ github, context } = {}) => {
   const newVersion = exec('npm', ['view', 'next', 'dist-tags.latest']);
   const branch = `next-${newVersion.replaceAll('.', '-')}`;
 
-  if (oldVersion === newVersion) {
-    // eslint-disable-next-line no-console
+  if (
+    github &&
+    exec('git', ['ls-remote', '--heads', 'origin', branch]).toString().trim()
+  ) {
+    console.log(`Branch ${branch} already exists, skipping update.`);
+    return;
+  }
+
+  let updatedCount = 0;
+
+  // update `examples/nextjs`
+  const oldVersion = require('../examples/nextjs/package.json').dependencies
+    .next;
+  if (github && oldVersion !== newVersion) {
+    updatedCount++;
+    exec('rm', ['-rf', './examples/nextjs']);
+    exec('npx', ['--yes', 'create-next-app@latest', './examples/nextjs']);
+    exec('git', [
+      'config',
+      '--global',
+      'user.email',
+      'infra+release@vercel.com',
+    ]);
+    exec('git', ['config', '--global', 'user.name', 'vercel-release-bot']);
+    exec('git', ['checkout', 'main']);
+    exec('git', ['checkout', '-b', branch]);
+    exec('git', ['add', '-A']);
+    exec('git', ['commit', '-m', branch]);
+    exec('git', ['push', 'origin', branch]);
+  }
+
+  // update tests
+  const testGlobs = [
+    'packages/cli/test/dev/fixtures/*/package.json',
+    'packages/cli/test/fixtures/unit/**/package.json',
+    'packages/fs-detectors/test/fixtures/**/package.json',
+    'packages/next/test/fixtures/*/package.json',
+    'packages/next/test/integration/*/package.json',
+    'packages/static-build/test/fixtures/*/package.json',
+  ];
+  for (const pattern of testGlobs) {
+    const files = await glob(pattern);
+    for (const file of files) {
+      const pkgJson = await readJson(file);
+      if (pkgJson.ignoreNextjsUpdates) {
+        continue;
+      }
+
+      const dependencies = [pkgJson.dependencies, pkgJson.devDependencies];
+      for (const deps of dependencies) {
+        const oldVersion = deps?.next;
+        if (oldVersion && oldVersion !== newVersion) {
+          updatedCount++;
+          deps.next = newVersion;
+          await writeJson(file, pkgJson, { spaces: 2 });
+
+          // update lock file
+          const cwd = dirname(file);
+          try {
+            if (await pathExists(join(cwd, 'yarn.lock'))) {
+              if (await pathExists(join(cwd, '.yarnrc.yml'))) {
+                console.warn(
+                  'Found .yarnrc.yml and this package probably uses Yarn v2/v3 which is not supported'
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** make message clear about where this file is‏

```suggestion
                  `Found ${join(cwd, '.yarnrc.yml')} and this package probably uses Yarn v2/v3 which is not supported`
```